spec/lib/granite/action/preconditions_spec.rb

```diff
@@ -101,4 +101,25 @@ def action_title
       specify { expect(Action.new(title: nil)).to satisfy_preconditions }
     end
   end
+
+  context 'with object' do
+    before do
+      stub_class(:test_precondition, Granite::Action::Precondition) do
+        def call(*)
+          decline_with(:wrong_title) if title != 'Ruby'
+        end
+      end
+
+      stub_class(:action, Granite::Action) do
+        attribute :title, String
+
+        precondition TestPrecondition
```

Comment:
It would be good to conditionally apply different precondition classes (depending on passed action arguments)...

Reply:
@bioform can you show an example about what do you mean?

Reply:
```ruby
precondition TestPrecondition, if: -> { title.present? }
```

Reply:
This already works, see [this](https://github.com/toptal/granite/pull/48/files#diff-40a9eaa41f33494fc7ecaf7f271023b6R30).